code/modules/antagonists/traitor/datum_mindslave.dm

```diff
@@ -21,10 +21,15 @@
 		return
 	master = _master
 	greet_text = _greet_text
-	return ..()
+	..()
 
-/datum/antagonist/mindslave/Destroy()
+/datum/antagonist/mindslave/Destroy(force, ...)
+	if(owner.som)
+		owner.som.serv -= owner
+		owner.som.leave_serv_hud(owner)
 	master = null
+	SSticker.mode.implanted[owner] = null
+	SSticker.mode.implanted -= owner
```

Comment:
might want to add this to the changelog as a fix. doesn't affect players, but it does affect admins.

Reply:
Added.